tests/_ast/test_app.py

```diff
@@ -732,6 +732,59 @@ def __():
         assert location is not None
         assert dirpath == location
 
+    def test_run_with_cli_args(self) -> None:
+        """Test that cli_args can be passed to app.run() and accessed via mo.cli_args()"""
+        app = App()
+
+        @app.cell
+        def __() -> tuple[Any]:
+            import marimo as mo
+            args = mo.cli_args()
+            return (args,)
+
+        # Test with various argument types
+        outputs, defs = app.run("--count", "42", "--name", "test", "--enabled", "true")
+
+        args = defs["args"]
+        assert args.get("count") == 42
+        assert args.get("name") == "test"
+        assert args.get("enabled") is True
```

Comment:
Is this just forwarding down to how we already handle CLI args? Where does the type handling come from?